crates/git_ui/src/worktree_picker.rs

```diff
@@ -0,0 +1,586 @@
+                })
+            })
+    }
+}
+
+#[derive(Debug, Clone)]
+struct WorktreeEntry {
+    worktree: GitWorktree,
+    positions: Vec<usize>,
+    is_new: bool,
+}
+
+pub struct WorktreeListDelegate {
+    matches: Vec<WorktreeEntry>,
+    all_worktrees: Option<Vec<GitWorktree>>,
+    workspace: WeakEntity<Workspace>,
+    repo: Option<Entity<Repository>>,
+    selected_index: usize,
+    last_query: String,
+    modifiers: Modifiers,
+    focus_handle: FocusHandle,
+    default_branch: Option<SharedString>,
+}
+
+impl WorktreeListDelegate {
+    fn new(
+        workspace: WeakEntity<Workspace>,
+        repo: Option<Entity<Repository>>,
+        _window: &mut Window,
+        cx: &mut Context<WorktreeList>,
+    ) -> Self {
+        Self {
+            matches: vec![],
+            all_worktrees: None,
+            workspace,
+            selected_index: 0,
+            repo,
+            last_query: Default::default(),
+            modifiers: Default::default(),
+            focus_handle: cx.focus_handle(),
+            default_branch: None,
+        }
+    }
+
+    fn create_worktree(
+        &self,
+        worktree_branch: &str,
+        replace_current_window: bool,
+        commit: Option<String>,
+        window: &mut Window,
+        cx: &mut Context<Picker<Self>>,
+    ) {
+        let workspace = self.workspace.clone();
+        let Some(repo) = self.repo.clone() else {
+            return;
+        };
+
+        let worktree_path = self
+            .workspace
+            .clone()
+            .update(cx, |this, cx| {
+                this.prompt_for_open_path(
+                    PathPromptOptions {
+                        files: false,
+                        directories: true,
+                        multiple: false,
+                        prompt: Some("Select directory for new worktree".into()),
+                    },
+                    DirectoryLister::Project(this.project().clone()),
+                    window,
+                    cx,
+                )
+            })
+            .log_err();
+        let Some(worktree_path) = worktree_path else {
+            return;
+        };
+
+        let branch = worktree_branch.to_string();
+        cx.spawn_in(window, async move |_, cx| {
+            let Some(paths) = worktree_path.await? else {
+                return anyhow::Ok(());
+            };
+            let path = paths.get(0).cloned().context("No path selected")?;
+
+            repo.update(cx, |repo, _| {
+                repo.create_worktree(branch.clone(), path.clone(), commit)
+            })?
+            .await??;
+
+            let final_path = path.join(branch);
+            workspace
+                .update_in(cx, |workspace, window, cx| {
+                    workspace.open_workspace_for_paths(
+                        replace_current_window,
+                        vec![final_path],
+                        window,
+                        cx,
+                    )
```

Comment:
This will always try to open the path in a local project, but when SSH remoting we want to open the path in a remote project.